apis/sfapi/volume.go

```diff
@@ -45,6 +45,48 @@ func (c *Client) GetVolumeByID(volID int64) (v Volume, err error) {
 	return volumes[0], nil
 }
 
+// GetVolumeByDockerName tbd
+func (c *Client) GetVolumeByDockerName(n string, acctID int64) (v Volume, err error) {
+	vols, err := c.GetVolumesByDockerName(n, acctID)
+	if err == nil && len(vols) == 1 {
+		return vols[0], nil
+	}
+
+	if len(vols) > 1 {
+		err = fmt.Errorf("Found more than one Volume with DockerName: %s for Account: %d", n, acctID)
+	} else if len(vols) < 1 {
+		err = fmt.Errorf("Failed to find any Volumes with DockerName: %s for Account: %d", n, acctID)
+	}
+	return v, err
+}
+
+// GetVolumesByDockerName tbd
+func (c *Client) GetVolumesByDockerName(dockerName string, acctID int64) (v []Volume, err error) {
+	var listReq ListVolumesForAccountRequest
+	var foundVolumes []Volume
+	listReq.AccountID = acctID
+	volumes, err := c.ListVolumesForAccount(&listReq)
+	if err != nil {
+		log.Error("Error retrieving volumes: ", err)
+		return foundVolumes, err
+	}
+	for _, vol := range volumes {
+		attrs, _ := vol.Attributes.(map[string]interface{})
+		//if vol.Attributes["docker-name"] == dockerName && vol.Status == "active" {
+		log.Infof("Looking for docker-name: %+v\n", attrs)
+		if attrs["docker-name"] == dockerName && vol.Status == "active" {
+			foundVolumes = append(foundVolumes, vol)
+		}
+	}
+	if len(foundVolumes) > 1 {
```

Comment:
I did convert the check on create to use the attributes so that shouldn't be a problem, BUT initially this len check was added because SF doesn't have any requirement that names are unique.  Now that we're using attributes this shouldn't be a problem, but I figure it's not a terrible thing to have in place just as a safety net.  We should "never" get there, but you know how that goes... as soon as you say that you end up finding yourself in that exact spot.